content/sensu-go/6.0/reference/filters.md

```diff
@@ -832,6 +832,96 @@ spec:
 
 {{< /language-toggle >}}
 
+## Use built-in JavaScript filter execution functions
+
+Sensu includes four built-in JavaScript functions for event filter execution:
+
+- `sensu.CheckDependencies`
+- `sensu.FetchEvent`
+- `sensu.CheckStatus`
+- `sensu.ListEvents`
+
+You can use these functions to query your event stores.
+
+### `sensu.CheckDependencies`
+
+The `sensu.CheckDependencies` function takes zero or more checks as arguments.
+It returns `true` if all the specified checks are passing or `false` if any of the specified checks are failing.
+
+If you do not specify any checks, the function always returns `true`.
+
+You can refer to the checks as strings:
+
+{{< code javascript >}}
+sensu.CheckDependencies("database", "disk")
+{{< /code >}}
+
```

Comment:
I'm thinking it would be better to provide these examples in an actual filter expression...

Reply:
So, to be clear, `sensu.CheckDependencies("database", "disk")` is a full filter expression on its own.

Reply:
The others are more complex. We don't intend for most people to use them, and are really only provided for tinkering.

Reply:
I'm sorry, I wasn't clear and also may not have been thinking about this clearly. I was wondering if we shouldn't provide a complete example filter definition that includes one of these functions as the expression.

Reply:
I added an example above of what I was thinking. Does this make sense at all? Or is it just more confusing?

Reply:
Your example makes sense but it would need quotes around the element in the expressions array, to be valid JSON.

Reply:
Sorry, my earlier comment was wrong.

No, this does not make sense. Sensu Query Expressions have to return a boolean. `sensu.FetchEvent` returns a complex data structure (an event).

You'd have to do something like
```
"expressions": [
  '(function () { var diskEvent = sensu.FetchEvent("router01", "disk"); if (diskEvent == nil) { return false; } return diskEvent.check.status == 0; })()'
]
```

Reply:
I don't believe these functions belong inlined in the Sensu Query Expressions docs. They are building blocks for creating Sensu Query Expressions (except for sensu.CheckDependencies, which is a Sensu Query Expression, because it returns a boolean)

Reply:
^ I have them in the event filter reference rather than the SQE reference. Is that what you mean?